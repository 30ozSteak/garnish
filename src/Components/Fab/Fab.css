.menu {
  bottom: 5%;
  cursor: pointer;
  height: 50px;
  margin: 0 auto;
  position: fixed;
  right: 20px;
  width: 50px;
  z-index: 1000;
}

.menu-button {
  background-color: var(--GFmint);
  background-repeat: no-repeat;
  background-size: 20px;
  box-shadow: 2px 2px 10px rgba(0, 0, 0, 0.459);
  height: 50px;
  right: 20px;
  width: 50px;
  position: fixed;
  transition: 0.5s;
  z-index: 1;
}

.menu a {
  background: var(--GFmint);
  height: 100%;
  left: 0;
  overflow: hidden;
  position: absolute;
  transform: scale(0);
  transition: 0.5s;
  width: 100%;
}

.menu a:hover {
  filter: brightness(110%);
  box-shadow: 2px 2px 10px rgba(0, 0, 0, 0.459);
}

.menu ul.active a {
  transform: scale(1);
}

.menu ul.active {
  height: 90px;
  width: 90px;
}

.menu ul.active a:nth-child(1) {
  background: center url(../../Styles/assets/home.png);
  background-color: var(--GFmint);
  background-repeat: no-repeat;
  background-size: 20px;
  filter: brightness(105%);
  transition-delay: 0s;
  top: -195%;
  z-index: 2;
}

.menu ul.active a:nth-child(2) {
  background: center url(../../Styles/assets/settings.svg);
  background-color: var(--GFmint);
  background-repeat: no-repeat;
  background-size: 20px;
  top: -295%;
  transition-delay: 0.2s;
  filter: brightness(110%);
  z-index: 1;
}

.menu ul.active a:nth-child(3) {
  background: center url(../../Styles/assets/star.svg);
  background-color: var(--GFmint);
  background-repeat: no-repeat;
  background-size: 20px;
  filter: brightness(115%);
  top: -395%;
  transition-delay: 0.4s;
  z-index: 0;
}

.cross {
  color: white;
  font-size: 2.5rem;
  height: 50px;
  width: 50px;
  margin: 0 auto;
  position: absolute;
  text-align: center;
  transition: 0.5s;
  z-index: 5;
}

.cross:hover {
  box-shadow: 2px 2px 10px rgba(0, 0, 0, 0.459);
}
