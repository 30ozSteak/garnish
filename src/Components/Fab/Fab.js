import React, { Component } from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import "./Fab.css";

export default class Menu extends Component {
  constructor(props, favorites) {
    super(props);
    this.state = {
      buttonState: ""
    };
  }

  toggleActive = () => {
    this.state.buttonState === ""
      ? this.setState({ buttonState: "active" })
      : this.setState({ buttonState: "" });
  };

  render() {
    return (
      <div className="menu">
        <div className="menu-button" onClick={this.toggleActive} />
        <ul className={this.state.buttonState}>
          <span onClick={this.toggleActive} className="cross">
            <p>+</p>
          </span>
          <div>
            <NavLink to="/" onClick={this.props.handlePeopleLink} />
            <NavLink to="/notes" onClick={this.props.handleNotesLink} />
            <NavLink to="/settings" onClick={this.props.handleSettingsLink} />
          </div>
        </ul>
      </div>
    );
  }
}
