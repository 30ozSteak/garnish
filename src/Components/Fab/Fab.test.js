import React from "react";
import Fab from "./Fab";
import { shallow } from "enzyme";

describe("Fab", () => {
  it("should resemble the snapshot", () => {
    const wrapper = shallow(<Fab />);
    expect(wrapper).toMatchSnapshot();
  });
});
