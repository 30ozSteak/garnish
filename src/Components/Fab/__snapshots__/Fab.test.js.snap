// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab should resemble the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="menu-button"
          onClick={[Function]}
        />,
        <ul
          className=""
        >
          <span
            className="cross"
            onClick={[Function]}
          >
            <p>
              +
            </p>
          </span>
          <div>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/settings"
            />
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/faves"
            />
          </div>
        </ul>,
      ],
      "className": "menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "menu-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="cross"
              onClick={[Function]}
            >
              <p>
                +
              </p>
            </span>,
            <div>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/settings"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/faves"
              />
            </div>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                +
              </p>,
              "className": "cross",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+",
              },
              "ref": null,
              "rendered": "+",
              "type": "p",
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/"
                />,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/settings"
                />,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/faves"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "to": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "to": "/settings",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "to": "/faves",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="menu-button"
            onClick={[Function]}
          />,
          <ul
            className=""
          >
            <span
              className="cross"
              onClick={[Function]}
            >
              <p>
                +
              </p>
            </span>
            <div>
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/settings"
              />
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/faves"
              />
            </div>
          </ul>,
        ],
        "className": "menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "menu-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="cross"
                onClick={[Function]}
              >
                <p>
                  +
                </p>
              </span>,
              <div>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/"
                />
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/settings"
                />
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  to="/faves"
                />
              </div>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  +
                </p>,
                "className": "cross",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "+",
                },
                "ref": null,
                "rendered": "+",
                "type": "p",
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/"
                  />,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/settings"
                  />,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/faves"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "to": "/settings",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "to": "/faves",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
